graphql/query_resolver.go

```diff
@@ -963,6 +963,61 @@ func (r *queryResolver) TaskNamesForBuildVariant(ctx context.Context, projectIde
 	return buildVariantTasks, nil
 }
 
+// Waterfall is the resolver for the waterfall field.
+func (r *queryResolver) Waterfall(ctx context.Context, options WaterfallOptions) (*Waterfall, error) {
+	projectId, err := model.GetIdForProject(options.ProjectIdentifier)
+	if err != nil {
+		return nil, ResourceNotFound.Send(ctx, fmt.Sprintf("finding project with id: '%s'", options.ProjectIdentifier))
+	}
+	limit := model.DefaultWaterfallVersionLimit
+	if utility.FromIntPtr(options.Limit) != 0 {
+		limit = utility.FromIntPtr(options.Limit)
+	}
+	requesters := options.Requesters
+	if len(requesters) == 0 {
+		requesters = evergreen.SystemVersionRequesterTypes
+	}
+
+	opts := model.WaterfallOptions{
+		Limit:      limit,
+		Requesters: requesters,
+	}
+
+	versions, err := model.GetWaterfallVersions(ctx, projectId, opts)
+	if err != nil {
+		return nil, ResourceNotFound.Send(ctx, fmt.Sprintf("getting versions: %s", err.Error()))
```

Comment:
Is this a not found or an internal server error? I would have thought that if the aggregation fails, it counts as an internal error.

Same question about the build variants.